File: professor.java
import java.io.File;
import java.util.Scanner;


public class professor {
   public static void main(String[] args) throws Exception {
      
      // Open file for input
      Scanner in = new Scanner(System.in);
      
      // Get number of test cases
      int testCases = in.nextInt();
      
      // Process each test case
      for (int caseNum = 1; caseNum <= testCases; caseNum++) {
         
         // Get the test case
         int stones = in.nextInt();
         int m = in.nextInt();
         
         // Print test case header
         System.out.printf("Game #%d: ", caseNum);
         
         // Let us remove a stone from the game, since the last stone is the losing one
         stones--;
         
         // Obvious, 0 stones is a losing state
         // 1 through m stones is a winning state because I can take all the stones
         // Thus m + 1 stones is also a losing state because it only goes to winning states
         // By repeating this logic, it's a losing state if the number of stones is divisible by m + 1
         // It is a winning state otherwise         
         if (stones % (m + 1) == 0) {
            System.out.println("Gentlemen don't cry.");
         }
         else {
            System.out.println("So long, Motchsteke!");
         }
         
      }
      
   }
}

File: professor.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
   int t, n, m, k, i;
   FILE* ifp = stdin;
   
   fscanf(ifp, "%d", &t);
   
   for (i = 0; i < t; i++)
   {
      fscanf(ifp, "%d %d", &n, &m);
      // Subtract one from the number of stones, game is now the player who cannot take a stone loses.
      n--;
      printf("Game #%d: ", i + 1);

      // 0 is a losing state.
      // 1 to m is a winning state, since the player can take up to m stones, and leave 0 for the next player.
      // m + 1 is a losing state, since no matter what happens, the player will leave 1 to m stones for the next player, which will be a winning state
      // for that player.
      // In general, if n is divisible by m + 1, the player is in a losing state, otherwise they are in a winning state.
      if (n % (m + 1) == 0)
         printf("Gentlemen don't cry.\n");
      else
         printf("So long, Motchsteke!\n");
   }
}
